### PR TITLE
docs: add rfc to describe cluster proxy/agent

### PR DESCRIPTION
**Pull request by ashuralyk**
@Flouse @ShookLyngs @quake anything recommended to change can be posted here

### CODE REVIEW

**Review by ShookLyngs — CHANGES_REQUESTED** [review]

**Review by linnnsss — COMMENTED** [review]
I recommend giving separate links/headings to each step and the associated methods, to enhance the overall structure with improved readability.

File: RFC_PROXY_AGENT.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        code_hash: CLUSTER_PROXY_TYPE_HASH
+	      args: <cluster_proxy_id> [<min_payment>]
+    Lock:
+        <user_defined>
+```
+
+The Type args can be:
+
+- args: <cluster_proxy_id>
+- args: <cluster_proxy_id> <minimal payment in 10^n ckybytes: uint16>
```
Comment (Flouse) [inline RFC_PROXY_AGENT.md:38]: 1. What is `ckybytes`?
2. @ShookLyngs said that type `uint16` should be `uint8`, right?

**Review by Flouse — CHANGES_REQUESTED** [review]

**Comment by ashuralyk** [comment]
> I recommend giving separate links/headings to each step and the associated methods, to enhance the overall structure with improved readability.

recommended changes have been applied, please review again

@linnnsss @Flouse @ShookLyngs

File: RFC_PROXY_AGENT.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+At this stage, the **owner** needs to create a special cell, here we called it **Cluster Proxy Cell**
+
+A Cluster Proxy Cell’s structure is like below:
+
+```yaml
+Cluster Proxy Cell:
+    Data: REFERENCED_CLUSTER_ID
+    Type:
+        code_hash: CLUSTER_PROXY_TYPE_HASH
+        args: <cluster_proxy_id> [<min_payment>]
```
Comment (Flouse) [inline RFC_PROXY_AGENT.md:28]: As we discussed, will `min_payment` be moved to `cell_data` in this RPC version or next version?

**Review by Flouse — APPROVED** [review]

File: RFC_PROXY_AGENT.md (hunk 2 above)
Comment (ashuralyk) [reply RFC_PROXY_AGENT.md:28]: after I complete it, I'll change this doc description about `min_payment`

**Review by linnnsss — APPROVED** [review]

**Pull request merged**